en/test_and_ci/continous_integration.md

```diff
@@ -1,3 +1,100 @@
+
+A number of additional artifacts are built when changes are made to the `main` branch or to `release/*` branches (any releases).
+These are then deployed to other repositories, uploaded to servers, and generally made available to other systems and services that use them.
+
+Release/main actions are managed by the [.github/workflows/deploy_all.yml](https://github.com/PX4/PX4-Autopilot/blob/main/.github/workflows/deploy_all.yml) workflow.
+
+For each board target, the artifacts built are:
+
+- Parameter metadata (`parameters.*`)
+- Events metadata (`events/*.xz`)
+- Actuators metadata (`actuators.json*`)
+- Target binary (`*.px4`)
+
+These files are uploaded to the [Amazon S3 bucket server](https://px4-travis.s3.amazonaws.com/).
+
+In addition, if there are changes to the `main` branch Jenkins builds / deploys metadata used by QGroundControl and documentation builds, based on the definitions in [PX4-Autopilot/Jenkinsfile](https://github.com/PX4/PX4-Autopilot/blob/main/Jenkinsfile).
+
+
+## PX4 Documentation
+
+Jenkins uses the [PX4BuildBot](https://github.com/PX4BuildBot) account to upload the following metadata to the [PX4-user_guide](https://github.com/PX4/PX4-user_guide) repo when the PX4-Autopilot `main` branch is altered.
+
+- `airframes_reference.md`: Markdown for [Airframes Reference Documentation](../airframes/airframe_reference.md)
+- `parameter_reference.md`: Markdown for [Parameter Reference Documentation](../advanced_config/parameter_reference.md)
+- `modules/*.md`: Markdown for [Modules documentation](../modules/README.md)
+- `graph_*.json`: JSON defining uORB Graph, visualized in [uORB Graph](../middleware/uorb_graph.md) page
+- `msg_docs/*.md`: Markdown for [uORB Messages documentation](../msg_docs/README.md)
+- `failsafe_sim/*`: Files for [Failsafe Simulator Interface](../config/safety_simulation.md)
+
+An example commit can be found [here](https://github.com/PX4/PX4-user_guide/commit/a4ffaf5fc32df8650da614aa3129b3b90794584f).
+
+The PX4 user guide is built when there are any changes to the [PX4-user_guide](https://github.com/PX4/PX4-user_guide) repo `main` or `v1.*` branches, including the addition of the metadata above.
+This ensures ensures that the "main" version of the documentation is always up to date.
+
+Note that the documentation build is also run using Jenkins CI, and is based on the definition in [PX4-user_guide/Jenkinsfile](https://github.com/PX4/PX4-user_guide/blob/main/Jenkinsfile).
+
+
+## QGroundControl 
+
+The PX4 firmware uploaded by `deploy_all.yml` is used by QGroundControl for [PX4 Firmware updates](../config/firmware.md) (stable: `release/*` / beta: `main`).
+ 
+The Jenkins server uses the [PX4BuildBot](https://github.com/PX4BuildBot) account to upload the following metadata to QGC whenever there is an update to main or a release branch:
+
+- `airframes.xml`: Airframes metadata describing all supported airframes
+- `parameters.xml`: Parameters metadata describing all supported parameters
```

Comment:
Correct, but should probably clarify that it copies the metadata into the QGC source tree.

Partially mentioned below, for parameters, most boards use the COMP_METADATA API to get it from the vehicle. And if not from the vehicle, then from S3 (which is pushed by the deploy_all.yml github action). Yes it's chaotic.

Reply:
Donee